hack/urllinter/pkg/lint/lint.go

```diff
@@ -0,0 +1,215 @@
+	}
+	llc.LinkMap = make(map[string][]LinkLint)
+	return llc, nil
+}
+
+func (llc *LinkLintConfig) Init(dir string) error {
+	err := filepath.Walk(dir,
+		func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			for _, exclude := range llc.ExcludePaths {
+				if strings.HasPrefix(path, exclude) {
+					return nil
+				} else if strings.HasPrefix(exclude, "*.") {
+					if filepath.Ext(path) == filepath.Ext(exclude) {
+						return nil
+					}
+				} else if string(exclude[len(exclude)-1]) != "/" { // its a file
+					if path == exclude {
+						return nil
+					}
+				}
+			}
+			ext := filepath.Ext(path)
+			for _, ex := range llc.IncludeExts {
+				if ext == ex {
+					err = llc.ReadFile(path)
+					if err != nil {
+						return err
+					}
+				}
+			}
+			return nil
+		})
+	if err != nil {
+		return err
+	}
+	return err
+}
+
+func (llc *LinkLintConfig) ReadFile(path string) error {
+	f, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	s := bufio.NewScanner(f)
+	count := 1
+	rxStrict := xurls.Strict()
+	for s.Scan() {
+		skip := false
+		line := strings.Trim(s.Text(), " ")
+		link := rxStrict.FindString(line)
+		col := strings.Index(s.Text(), link)
+		// ignore lines
+		// if the line is commented then skip it
+		// This is for go or programming comments only
+
+		// if len(line) >= 2 && line[:2] == "//" {
+		// 	continue
+		// }
+		// // comments for yaml or yml files. Do not consider that line if line start with a comment
+		// if len(line) > 1 && line[:1] == "#" {
+		// 	continue
+		// }
+		// // comments for yaml or yml files. If there is comment in the line take only uncommented part
+		// index := strings.Index(line, "#")
+		// if index > 0 {
+		// 	line = line[0:index]
+		// }
+		// // This is for go or programming code only as comments in yaml files start with #
+		// // start
+		// if len(line) >= 2 && line[:2] == "/*" {
+		// 	//TODO here
+		// 	skip = true
+		// }
+		// if strings.Contains(line, "*/") {
+		// 	skip = false
+		// }
+		// if skip {
+		// 	continue
+		// }
+
+		if len(link) >= 8 && strings.ToLower(strings.Trim(link, " ")[0:4]) != "http" { // do not consider it as url if it dies not start with http or https
+			continue
+		}
+		for _, l := range llc.ExcludeLinks {
+			if strings.Contains(link, l) {
+				skip = true
+				break
+			}
+		}
+		if link != "" && !skip {
+			llints := llc.LinkMap[link]
+			llc.LinkMap[link] = append(llints, LinkLint{Path: path, Line: link, Position: Position{Row: count, Col: col}, Status: "", Message: ""})
+		}
+		count++
+	}
+	err = s.Err()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (llc *LinkLintConfig) LintAll() bool {
+	isFatal := false
+	count := 0
+	for key := range llc.LinkMap {
+		count++
+		fmt.Println("Currently checking ", count, " url(s) out of ", len(llc.LinkMap))
+		if !IsURL(key) {
+			isFatal = true
+			llc.OnFail("Invalid URL", key)
+			continue
+		}
+		cli := &http.Client{}
```

Comment:
For #2948 

We need to introduce a configuration in `http.Client{}` like this -

```golang
tr := &http.Transport{
    TLSClientConfig: {
      InsecureSkipVerify: true
    }
}
cli := &http.Client{Transport: tr}
```

References - 
- https://pkg.go.dev/net/http#pkg-overview
- https://pkg.go.dev/net/http#Transport
- https://pkg.go.dev/crypto/tls#Config

Reply:
cc @aman556

Reply:
I have triggered the URL linter after adding the failing URL once it complete i will try the above suggestion and check that it is working or not.

Reply:
after adding '"crypto/tls"` library and adding this
```tr := &http.Transport{
        TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
    	}

		cli := &http.Client{Transport: tr}```
issue got resolved. I am updating this PR.